### PR TITLE
Add Model Tests for several models

### PR DESCRIPTION
**Pull request by manjotsidhu**
This Pull Request, adds model test for all the various models that needed proper unit testing.
- [x] AddCollaboratorResponse
- [x] AddGroupMembersResponse
- [x] Grade
- [x] Links
- [x] Collaborators
- [x] Projects
- [x] Assignments
- [x] GroupMembers
- [x] Groups

### CODE REVIEW

**Comment by manjotsidhu** [comment]
@Nitish145 Please review this pull request.

File: test/model_tests/add_collaborator_response_test.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mobile_app/models/add_collaborator_response.dart';
+
+import '../setup/test_data/mock_add_collaborators_response.dart';
+
+void main() {
+  group('AddCollaboratorResponseTest -', () {
+    test('fromJson', () {
+      var _collaborator_response =
```
Comment (Nitish145) [inline test/model_tests/add_collaborator_response_test.dart:9]: Use camelCase.

File: test/model_tests/add_group_members_response_test.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mobile_app/models/add_group_members_response.dart';
+
+import '../setup/test_data/mock_add_group_members_response.dart';
+
+void main() {
+  group('AddGroupMembersResponseTest -', () {
+    test('fromJson', () {
+      var _collaborator_response =
```
Comment (Nitish145) [inline test/model_tests/add_group_members_response_test.dart:9]: Use camelCase string nomenclature.

File: test/model_tests/assignments_test.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    test('AssignmentAttributesTest', () {
+      var _assignmentAttributes =
+          AssignmentAttributes.fromJson(mockAssignmentAttributes);
+
+      expect(_assignmentAttributes, isInstanceOf<AssignmentAttributes>());
+
+      expect(_assignmentAttributes.name, 'Test');
+      expect(_assignmentAttributes.deadline,
+          DateTime.parse('2020-08-15T15:23:00.000Z').toLocal());
```
Comment (Nitish145) [inline test/model_tests/assignments_test.dart:48]: `toLocal` can potentially result in different results in different timezones of Continous Integration.

File: test/model_tests/assignments_test.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      expect(_assignmentAttributes.description, 'description');
+      expect(_assignmentAttributes.status, 'open');
+      expect(_assignmentAttributes.hasMentorAccess, true);
+      expect(_assignmentAttributes.createdAt,
+          DateTime.parse('2020-08-19T15:00:30.423Z').toLocal());
```
Comment (Nitish145) [inline test/model_tests/assignments_test.dart:53]: `toLocal`

File: test/model_tests/assignments_test.dart [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      expect(_assignmentAttributes.updatedAt,
+          DateTime.parse('2020-08-20T15:23:03.018Z').toLocal());
```
Comment (Nitish145) [inline test/model_tests/assignments_test.dart:55]: `toLocal`

**Review by Nitish145 — CHANGES_REQUESTED** [review]

File: test/model_tests/assignments_test.dart (hunk 3 above)
Comment (manjotsidhu) [reply test/model_tests/assignments_test.dart:48]: But the modal logic converts date in Local time using `toLocal()` that's why I used `toLocal()` for maintaining consistency.

File: test/model_tests/assignments_test.dart (hunk 4 above)
Comment (manjotsidhu) [reply test/model_tests/assignments_test.dart:53]: modal uses toLocal().

File: test/model_tests/assignments_test.dart (hunk 5 above)
Comment (manjotsidhu) [reply test/model_tests/assignments_test.dart:55]: modal uses `toLocal()`

File: test/model_tests/assignments_test.dart (hunk 3 above)
Comment (Nitish145) [reply test/model_tests/assignments_test.dart:48]: Okay

**Review by Nitish145 — APPROVED** [review]

**Pull request merged**